### PR TITLE
feat: Add gitlab_repository_file data source

### PR DESCRIPTION
**Pull request by petrmvala**
This data source is part of Gitlab API.

Closes: #927

### PR Checklist Acknowledgement

- [ ] I acknowledge that all of the following items are true, where applicable:
  - Resource attributes match 1:1 the names and structure of the API resource in [the GitLab API documentation](https://docs.gitlab.com/ee/api/).
  - [Examples](https://github.com/gitlabhq/terraform-provider-gitlab/tree/main/examples) are updated with:
    - A \*.tf file for the resource/s with at least one usage example
    - A \*.sh file for the resource/s with an import example (if applicable)
  - New resources have at minimum a basic test with three steps:
    - Create the resource
    - Update the attributes
    - Import the resource
  - No new `//lintignore` comments were copied from existing code. (Linter rules are meant to be enforced on new code.)

### CODE REVIEW

**Review by timofurrer — COMMENTED** [review]
Could you give this approach a shot:

https://github.com/gitlabhq/terraform-provider-gitlab/blob/main/internal/provider/data_source_gitlab_project_issue.go

For that to work you may also add the additional read-only fields to the `gitlab_repository_file` resource as `Computed`. 

If you don't want I can later or tomorrow give it a shot, no worries. 

Also, the examples are missing ;)

**Comment by petrmvala** [comment]
Well, I posted this because I was not sure why does this error out and wanted to ask for advice https://github.com/gitlabhq/terraform-provider-gitlab/runs/5461795214?check_suite_focus=true#step:6:34

But the approach with the schema seems interesting.

**Comment by timofurrer** [comment]
@petrmvala It's unrelated, but probably related to #820 ...

File: internal/provider/data_source_gitlab_repository_file.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	gitlab "github.com/xanzy/go-gitlab"
+)
+
+var _ = registerDataSource("gitlab_repository_file", func() *schema.Resource {
+	return &schema.Resource{
+		Description: "Allows you to receive information about file in repository like name, size, content. File content is Base64 encoded. This endpoint can be accessed without authentication if the repository is publicly accessible.",
```
Comment (timofurrer) [inline internal/provider/data_source_gitlab_repository_file.go:15]: Please use the same description pattern as here:

https://github.com/gitlabhq/terraform-provider-gitlab/blob/c6a706e3ab6e907e2da0c92ee50138544ff501dd/internal/provider/data_source_gitlab_project_issue.go#L13

w.r.t to wording and the upstream docs link

File: internal/provider/data_source_gitlab_repository_file_test.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+		file := s.RootModule().Resources[src]
+		fileAttr := file.Primary.Attributes
+
+		search := s.RootModule().Resources[n]
+		searchAttr := search.Primary.Attributes
+
+		testAttributes := []string{
+			"project",
+			"file_path",
```
Comment (timofurrer) [inline internal/provider/data_source_gitlab_repository_file_test.go:39]:
```suggestion
```

**Review by timofurrer — CHANGES_REQUESTED** [review]
Looks good, I think we leave the things with `datasourceSchemaFromResourceSchema()` for now and go with the proposed approach.

File: internal/provider/data_source_gitlab_repository_file.go (hunk 1 above)
Comment (petrmvala) [reply internal/provider/data_source_gitlab_repository_file.go:15]: I will take a look again probably on Sunday. I had to actually implement something at my workplace :D

**Comment by timofurrer** [comment]
@petrmvala I took the liberty and followed up on this.

I'll merge once the checks passed!

Thank you very much for the contribution! 🎉

**Review by timofurrer — APPROVED** [review]

**Pull request merged**